beacon-light-client/circom/scripts/aggregate_bitmask/aggregate_bitmask.sh

```diff
@@ -1,78 +1,44 @@
 #!/usr/bin/env bash
+source "${BASH_SOURCE%/*}/../../../../scripts/utils/paths.sh"
+source "${BASH_SOURCE%/*}/../common.sh"
 
-PHASE1=../../circuits/pot25_final.ptau
+CIRCOM_DIR="${ROOT}/beacon-light-client/circom"
+SNARKJS_DIR="${ROOT}"/vendor/snarkjs
+SNARKJS="${SNARKJS_DIR}"/cli.js
+PHASE1=~/Downloads/powersOfTau28_hez_final_24.ptau
 CIRCUIT_NAME=aggregate_bitmask
-BUILD_DIR=../../build/"$CIRCUIT_NAME"
+BUILD_DIR="${CIRCOM_DIR}/build/${CIRCUIT_NAME}"
+CIRCUIT_DIR="${CIRCOM_DIR}/scripts/${CIRCUIT_NAME}"
 
-if [ -f "$PHASE1" ]; then
-    echo "Found Phase 1 ptau file"
-else
-    echo "No Phase 1 ptau file found. Exiting..."
-    exit 1
-fi
+git submodule update --init --recursive
 
-if [ ! -d "$BUILD_DIR" ]; then
-    echo "No build directory found. Creating build directory..."
-    mkdir -p "$BUILD_DIR"
-fi
+look_for_ptau_file
```

Comment:
Please use *parentheses* around the `cd` commands and in every function call that changes the directory. This ensures that even if the script crashes or exits prematurely, the directory changes are contained within a subshell, preventing unintended changes to the current working directory of the parent shell. You can find more information about command grouping in the [Bash Manual](https://www.gnu.org/savannah-checkouts/gnu/bash/manual/bash.html#Command-Grouping).

```suggestion
( look_for_ptau_file $PARAMx)
```

Reply:
Done